crates/autopilot/src/run_loop.rs

```diff
@@ -58,9 +61,54 @@ pub struct RunLoop {
     /// Maintenance tasks that should run before every runloop to have
     /// the most recent data available.
     pub maintenance: Arc<Maintenance>,
+    /// Queue for executing settle futures one by one guarantying FIFO execution
+    /// order.
+    execution_queue: mpsc::UnboundedSender<BoxFuture<'static, ()>>,
 }
 
 impl RunLoop {
+    #[allow(clippy::too_many_arguments)]
+    pub fn new(
+        eth: infra::Ethereum,
+        persistence: infra::Persistence,
+        drivers: Vec<Arc<infra::Driver>>,
+        solvable_orders_cache: Arc<SolvableOrdersCache>,
+        market_makable_token_list: AutoUpdatingTokenList,
+        submission_deadline: u64,
+        max_settlement_transaction_wait: Duration,
+        solve_deadline: Duration,
+        liveness: Arc<Liveness>,
+        synchronization: RunLoopMode,
+        max_run_loop_delay: Duration,
+        maintenance: Arc<Maintenance>,
+    ) -> Self {
+        let (tx, mut rx) = mpsc::unbounded_channel();
+
+        tokio::spawn(async move {
+            while let Some(future) = rx.recv().await {
+                future.await;
+            }
+            tracing::info!("runloop execution queue stopped")
+        });
```

Comment:
I'm not super fond of this logic. This could theoretically cause solutions to pile up. For example when there are a bunch of solutions after another that each take a few blocks to submit. I think what might be a bit more desirable is storing a map `solver => task::JoinHandle`. If the winning solver does not already have a solution in-flight spawn the submission task and store it in the map.
If there is still a submission in flight await that first and block the run loop until the previous one finished before submitting the new one. 🤔

Reply:
Totally makes sense, thanks. Updated the code.